src/tree.rs

```diff
@@ -10,14 +10,14 @@ where
     A: Clone,
 {
     thunk: Lazy<'a, A>,
```

Comment:
do you reckon you get much out of the value itself being lazy? I expected that this would always end up being forced

Reply:
I feel like I had a version that was similar to yours and it didn't absolve the issue with `sequence`. Would need to consider truly getting rid of the need for it (which your port seems to have).